frontend/src/app/chart/chart.component.spec.ts

```diff
@@ -30,7 +30,7 @@ const getMaxMin = (records: number[]) => {
   ];
 };
 
-const generateNewData = (length: number) => {
+const generateSingleChannelData = (length: number) => {
   const recordsData: [number, number][] = [];
   for (let index = 0; index < length; index++) {
     recordsData.push([1573149236356988 + index * 2000, Math.random() * 500]);
```

Comment:
Since you are using this calculation multiple times, maybe it's better to abstract it as consts or methods, also please add some comments for why you are calculating in this way. I remembered that I asked this question before, but I already forgot [cry face], so some comments could help a lot.

Reply:
Refactored with a new function, and added comments on this.